### PR TITLE
Include more information in WINDOW_UPDATE frame logs

### PR DESCRIPTION
**Pull request by swankjesse**
As a part of investigating this issue:
https://github.com/square/okhttp/issues/7248

### CODE REVIEW

File: okhttp/src/jvmTest/java/okhttp3/internal/http2/FrameLogTest.kt [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+import okhttp3.internal.http2.Http2.TYPE_PING
+import okhttp3.internal.http2.Http2.TYPE_PUSH_PROMISE
+import okhttp3.internal.http2.Http2.TYPE_SETTINGS
+import okhttp3.internal.http2.Http2.formatFlags
+import okhttp3.internal.http2.Http2.frameLog
+import okhttp3.internal.http2.Http2.frameLogWindowUpdate
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class FrameLogTest {
```
Comment (swankjesse) [inline okhttp/src/jvmTest/java/okhttp3/internal/http2/FrameLogTest.kt:35]: Automatically converted to Kotlin

File: okhttp/src/jvmTest/java/okhttp3/internal/http2/FrameLogTest.kt [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      .isEqualTo("<< 0x00000003    22 HEADERS       END_HEADERS")
+    assertThat(frameLog(true, 3, 226, TYPE_DATA, FLAG_END_STREAM))
+      .isEqualTo("<< 0x00000003   226 DATA          END_STREAM")
+    assertThat(frameLog(false, 0, 8, TYPE_GOAWAY, FLAG_NONE))
+      .isEqualTo(">> 0x00000000     8 GOAWAY        ")
+  }
+
+  /** Window update frames have special formatting.  */
+  @Test
+  fun windowUpdateFrames() {
```
Comment (swankjesse) [inline okhttp/src/jvmTest/java/okhttp3/internal/http2/FrameLogTest.kt:61]: New test function

File: okhttp/src/jvmMain/kotlin/okhttp3/internal/http2/Http2Reader.kt [hunk 3]
```diff
@@ -308,6 +311,14 @@ class Http2Reader(
     if (length != 4) throw IOException("TYPE_WINDOW_UPDATE length !=4: $length")
```
Comment (yschimke) [inline okhttp/src/jvmMain/kotlin/okhttp3/internal/http2/Http2Reader.kt:311]: It seems a shame to not log the faulty frame in these two exception cases. Can we assume a rule that we intend to log each and every frame that comes in or goes out?

**Review by yschimke — APPROVED** [review]
Looks good.

File: okhttp/src/jvmMain/kotlin/okhttp3/internal/http2/Http2Reader.kt (hunk 3 above)
Comment (swankjesse) [reply okhttp/src/jvmMain/kotlin/okhttp3/internal/http2/Http2Reader.kt]: Fixed.

**Pull request merged**